Sources/TrimBehavior.swift

```diff
@@ -0,0 +1,86 @@
+//
+//  File.swift
+//  
+//
+//  Created by Yonas Kolb on 12/11/19.
+//
+
+import Foundation
+
+public struct TrimBehavior: Equatable {
+  var leading: Trim
+  var trailing: Trim
+
+  public enum Trim: Equatable {
+    /// nothing
+    case none
+
+    /// tabs and spaces
+    case whitespace
+
+    /// tabs and spaces and a single new line
+    case whitespaceAndOneNewLine
```

Comment:
I'm not sure if there is a better way but this case feels like it only applies to trailing behaviour. maybe two separate enums?